src/renderer/services/wallets/WalletInterfaces.ts

```diff
@@ -0,0 +1,29 @@
+import { V3Keystore, V3Params } from './eth1/Eth1KeystoreInterfaces';
+import { PrivateKey } from '@chainsafe/bls/lib/privateKey';
+import { BLSPubkey, BLSSecretKey, BLSSignature } from '@chainsafe/bls/lib/types';
+import { PublicKey } from '@chainsafe/bls/lib/publicKey';
+import { Keypair } from '@chainsafe/bls/lib/keypair';
+import { Hash } from '@chainsafe/eth2.0-types';
+
+export type IKeystoreJson = V3Keystore;
+export type KeystoreParams = V3Params;
+
+export type Domain = Buffer;
+
+export interface IKeystore {
```

Comment:
I would suggest interface like this:
```javascript
export interface ICGKeystore {
   new(file: string);
   decrypt(password: string): Keypair;
   changePassword(oldPassword: string, newPassword: string): void;
   //deletes physical keystore file
   destroy(): void;
   static create(file: string, password: string, keypair: Keypair): Keystore;
}
```
@aostrun you can assume same keystore interface

you also need Eth2HDWallet class like:
```javascript
export interface IEth2HDWallet {
   // should return mnemonic
   static generate(entropy?: bytes): string;

   static getKeypair(mnemonic: string, walletIndex?: number = 0): Keypair;
}
```

Reply:
Can we use something like this for the `ICGKeystore` as it will be easier to create interface implementations, problem is with the constructor signature  `new` ([here is the reason why](https://www.typescriptlang.org/docs/handbook/interfaces.html#difference-between-the-static-and-instance-sides-of-classes)).
```javascript
export interface ICGKeystore {
    decrypt(password: string): Keypair;
    changePassword(oldPassword: string, newPassword: string): void;
    destroy(): void;
    create(file: string, password: string, keypair: Keypair): ICGKeystore;
}

export interface ICGKeystoreFactory {
    new (file: string): ICGKeystore;
}
```